src/components/Textarea/Textarea.tsx

```diff
@@ -9,17 +9,41 @@ type Props = TextareaHTMLAttributes<HTMLTextAreaElement> & {
   autoFocus?: boolean
 }
 
-export const Textarea: FC<Props> = ({ autoFocus, ...props }) => {
+export const Textarea: FC<Props> = ({ autoFocus, maxLength, ...props }) => {
   const theme = useTheme()
   const ref = useRef<HTMLTextAreaElement>(null)
+  const [count, setCount] = useState(0)
 
   useEffect(() => {
     if (autoFocus && ref && ref.current) {
       ref.current.focus()
     }
   }, [autoFocus])
 
-  return <StyledTextarea {...props} ref={ref} themes={theme} />
+  const handleKeyup = (event: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    const value = event.currentTarget.value
+    setCount(value.length - (value.split(/[\uD800-\uDBFF][\uDC00-\uDFFF]/g).length - 1))
```

Comment:
[IMO] Explicitly making it variable makes it easier for those who see it later to understand that it is the range of the surrogate 💡

Reply:
This was necessary to support emoji in the defaultValue as well, so I summarized the process as a function.